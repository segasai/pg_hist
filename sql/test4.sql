select * from pg_hist_2d('select b,(c::numeric)%100 from test2',ARRAY[30,30],ARRAY[0,0],ARRAY[100,100]);
