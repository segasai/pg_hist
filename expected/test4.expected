 x1 | x2 | hist 
----+----+------
  0 |  0 |  248
  1 |  0 |  186
  2 |  0 |  186
  3 |  0 |  155
  4 |  0 |   93
  5 |  0 |   93
  6 |  0 |  124
  7 |  0 |   93
  8 |  0 |   93
  9 |  0 |  124
 10 |  0 |   93
 11 |  0 |   93
 12 |  0 |  124
 13 |  0 |   93
 14 |  0 |   93
 15 |  0 |  124
 16 |  0 |   93
 17 |  0 |   93
 18 |  0 |  124
 19 |  0 |   93
 20 |  0 |   93
 21 |  0 |  124
 22 |  0 |   93
 23 |  0 |   93
 24 |  0 |  124
 25 |  0 |   93
 26 |  0 |   93
 27 |  0 |  124
 28 |  0 |   93
 29 |  0 |   93
  0 |  1 |  124
  1 |  1 |   93
  2 |  1 |   93
  3 |  1 |  217
  4 |  1 |  186
  5 |  1 |  186
  6 |  1 |  124
  7 |  1 |   93
  8 |  1 |   93
  9 |  1 |  124
 10 |  1 |   93
 11 |  1 |   93
 12 |  1 |  124
 13 |  1 |   93
 14 |  1 |   93
 15 |  1 |  124
 16 |  1 |   93
 17 |  1 |   93
 18 |  1 |  124
 19 |  1 |   93
 20 |  1 |   93
 21 |  1 |  124
 22 |  1 |   93
 23 |  1 |   93
 24 |  1 |  124
 25 |  1 |   93
 26 |  1 |   93
 27 |  1 |  124
 28 |  1 |   93
 29 |  1 |   93
  0 |  2 |  124
  1 |  2 |   93
  2 |  2 |   93
  3 |  2 |  124
  4 |  2 |   93
  5 |  2 |   93
  6 |  2 |  248
  7 |  2 |  186
  8 |  2 |  186
  9 |  2 |  124
 10 |  2 |   93
 11 |  2 |   93
 12 |  2 |  124
 13 |  2 |   93
 14 |  2 |   93
 15 |  2 |  124
 16 |  2 |   93
 17 |  2 |   93
 18 |  2 |  124
 19 |  2 |   93
 20 |  2 |   93
 21 |  2 |  124
 22 |  2 |   93
 23 |  2 |   93
 24 |  2 |  124
 25 |  2 |   93
 26 |  2 |   93
 27 |  2 |  124
 28 |  2 |   93
 29 |  2 |   93
  0 |  3 |  124
  1 |  3 |   93
  2 |  3 |   93
  3 |  3 |  124
  4 |  3 |   93
  5 |  3 |   93
  6 |  3 |  124
  7 |  3 |   93
  8 |  3 |   93
  9 |  3 |  248
 10 |  3 |  186
 11 |  3 |  186
 12 |  3 |  155
 13 |  3 |   93
 14 |  3 |   93
 15 |  3 |  124
 16 |  3 |   93
 17 |  3 |   93
 18 |  3 |  124
 19 |  3 |   93
 20 |  3 |   93
 21 |  3 |  124
 22 |  3 |   93
 23 |  3 |   93
 24 |  3 |  124
 25 |  3 |   93
 26 |  3 |   93
 27 |  3 |  124
 28 |  3 |   93
 29 |  3 |   93
  0 |  4 |  124
  1 |  4 |   93
  2 |  4 |   93
  3 |  4 |  124
  4 |  4 |   93
  5 |  4 |   93
  6 |  4 |  124
  7 |  4 |   93
  8 |  4 |   93
  9 |  4 |  124
 10 |  4 |   93
 11 |  4 |   93
 12 |  4 |  217
 13 |  4 |  186
 14 |  4 |  186
 15 |  4 |  124
 16 |  4 |   93
 17 |  4 |   93
 18 |  4 |  124
 19 |  4 |   93
 20 |  4 |   93
 21 |  4 |  124
 22 |  4 |   93
 23 |  4 |   93
 24 |  4 |  124
 25 |  4 |   93
 26 |  4 |   93
 27 |  4 |  124
 28 |  4 |   93
 29 |  4 |   93
  0 |  5 |  124
  1 |  5 |   93
  2 |  5 |   93
  3 |  5 |  124
  4 |  5 |   93
  5 |  5 |   93
  6 |  5 |  124
  7 |  5 |   93
  8 |  5 |   93
  9 |  5 |  124
 10 |  5 |   93
 11 |  5 |   93
 12 |  5 |  124
 13 |  5 |   93
 14 |  5 |   93
 15 |  5 |  248
 16 |  5 |  186
 17 |  5 |  186
 18 |  5 |  124
 19 |  5 |   93
 20 |  5 |   93
 21 |  5 |  124
 22 |  5 |   93
 23 |  5 |   93
 24 |  5 |  124
 25 |  5 |   93
 26 |  5 |   93
 27 |  5 |  124
 28 |  5 |   93
 29 |  5 |   93
  0 |  6 |  124
  1 |  6 |   93
  2 |  6 |   93
  3 |  6 |  124
  4 |  6 |   93
  5 |  6 |   93
  6 |  6 |  124
  7 |  6 |   93
  8 |  6 |   93
  9 |  6 |  124
 10 |  6 |   93
 11 |  6 |   93
 12 |  6 |  124
 13 |  6 |   93
 14 |  6 |   93
 15 |  6 |  124
 16 |  6 |   93
 17 |  6 |   93
 18 |  6 |  248
 19 |  6 |  186
 20 |  6 |  186
 21 |  6 |  155
 22 |  6 |   93
 23 |  6 |   93
 24 |  6 |  124
 25 |  6 |   93
 26 |  6 |   93
 27 |  6 |  124
 28 |  6 |   93
 29 |  6 |   93
  0 |  7 |  124
  1 |  7 |   93
  2 |  7 |   93
  3 |  7 |  124
  4 |  7 |   93
  5 |  7 |   93
  6 |  7 |  124
  7 |  7 |   93
  8 |  7 |   93
  9 |  7 |  124
 10 |  7 |   93
 11 |  7 |   93
 12 |  7 |  124
 13 |  7 |   93
 14 |  7 |   93
 15 |  7 |  124
 16 |  7 |   93
 17 |  7 |   93
 18 |  7 |  124
 19 |  7 |   93
 20 |  7 |   93
 21 |  7 |  217
 22 |  7 |  186
 23 |  7 |  186
 24 |  7 |  124
 25 |  7 |   93
 26 |  7 |   93
 27 |  7 |  124
 28 |  7 |   93
 29 |  7 |   93
  0 |  8 |  124
  1 |  8 |   93
  2 |  8 |   93
  3 |  8 |  124
  4 |  8 |   93
  5 |  8 |   93
  6 |  8 |  124
  7 |  8 |   93
  8 |  8 |   93
  9 |  8 |  124
 10 |  8 |   93
 11 |  8 |   93
 12 |  8 |  124
 13 |  8 |   93
 14 |  8 |   93
 15 |  8 |  124
 16 |  8 |   93
 17 |  8 |   93
 18 |  8 |  124
 19 |  8 |   93
 20 |  8 |   93
 21 |  8 |  124
 22 |  8 |   93
 23 |  8 |   93
 24 |  8 |  248
 25 |  8 |  186
 26 |  8 |  186
 27 |  8 |  124
 28 |  8 |   93
 29 |  8 |   93
  0 |  9 |  151
  1 |  9 |   90
  2 |  9 |   90
  3 |  9 |  120
  4 |  9 |   90
  5 |  9 |   90
  6 |  9 |  120
  7 |  9 |   90
  8 |  9 |   90
  9 |  9 |  120
 10 |  9 |   90
 11 |  9 |   90
 12 |  9 |  120
 13 |  9 |   90
 14 |  9 |   90
 15 |  9 |  120
 16 |  9 |   90
 17 |  9 |   90
 18 |  9 |  120
 19 |  9 |   90
 20 |  9 |   90
 21 |  9 |  120
 22 |  9 |   90
 23 |  9 |   90
 24 |  9 |  120
 25 |  9 |   90
 26 |  9 |   90
 27 |  9 |  244
 28 |  9 |  183
 29 |  9 |  183
  0 | 10 |  210
  1 | 10 |  180
  2 | 10 |  180
  3 | 10 |  120
  4 | 10 |   90
  5 | 10 |   90
  6 | 10 |  120
  7 | 10 |   90
  8 | 10 |   90
  9 | 10 |  120
 10 | 10 |   90
 11 | 10 |   90
 12 | 10 |  120
 13 | 10 |   90
 14 | 10 |   90
 15 | 10 |  120
 16 | 10 |   90
 17 | 10 |   90
 18 | 10 |  120
 19 | 10 |   90
 20 | 10 |   90
 21 | 10 |  120
 22 | 10 |   90
 23 | 10 |   90
 24 | 10 |  120
 25 | 10 |   90
 26 | 10 |   90
 27 | 10 |  120
 28 | 10 |   90
 29 | 10 |   90
  0 | 11 |  120
  1 | 11 |   90
  2 | 11 |   90
  3 | 11 |  240
  4 | 11 |  180
  5 | 11 |  180
  6 | 11 |  120
  7 | 11 |   90
  8 | 11 |   90
  9 | 11 |  120
 10 | 11 |   90
 11 | 11 |   90
 12 | 11 |  120
 13 | 11 |   90
 14 | 11 |   90
 15 | 11 |  120
 16 | 11 |   90
 17 | 11 |   90
 18 | 11 |  120
 19 | 11 |   90
 20 | 11 |   90
 21 | 11 |  120
 22 | 11 |   90
 23 | 11 |   90
 24 | 11 |  120
 25 | 11 |   90
 26 | 11 |   90
 27 | 11 |  120
 28 | 11 |   90
 29 | 11 |   90
  0 | 12 |  120
  1 | 12 |   90
  2 | 12 |   90
  3 | 12 |  120
  4 | 12 |   90
  5 | 12 |   90
  6 | 12 |  240
  7 | 12 |  180
  8 | 12 |  180
  9 | 12 |  150
 10 | 12 |   90
 11 | 12 |   90
 12 | 12 |  120
 13 | 12 |   90
 14 | 12 |   90
 15 | 12 |  120
 16 | 12 |   90
 17 | 12 |   90
 18 | 12 |  120
 19 | 12 |   90
 20 | 12 |   90
 21 | 12 |  120
 22 | 12 |   90
 23 | 12 |   90
 24 | 12 |  120
 25 | 12 |   90
 26 | 12 |   90
 27 | 12 |  120
 28 | 12 |   90
 29 | 12 |   90
  0 | 13 |  120
  1 | 13 |   90
  2 | 13 |   90
  3 | 13 |  120
  4 | 13 |   90
  5 | 13 |   90
  6 | 13 |  120
  7 | 13 |   90
  8 | 13 |   90
  9 | 13 |  210
 10 | 13 |  180
 11 | 13 |  180
 12 | 13 |  120
 13 | 13 |   90
 14 | 13 |   90
 15 | 13 |  120
 16 | 13 |   90
 17 | 13 |   90
 18 | 13 |  120
 19 | 13 |   90
 20 | 13 |   90
 21 | 13 |  120
 22 | 13 |   90
 23 | 13 |   90
 24 | 13 |  120
 25 | 13 |   90
 26 | 13 |   90
 27 | 13 |  120
 28 | 13 |   90
 29 | 13 |   90
  0 | 14 |  120
  1 | 14 |   90
  2 | 14 |   90
  3 | 14 |  120
  4 | 14 |   90
  5 | 14 |   90
  6 | 14 |  120
  7 | 14 |   90
  8 | 14 |   90
  9 | 14 |  120
 10 | 14 |   90
 11 | 14 |   90
 12 | 14 |  240
 13 | 14 |  180
 14 | 14 |  180
 15 | 14 |  120
 16 | 14 |   90
 17 | 14 |   90
 18 | 14 |  120
 19 | 14 |   90
 20 | 14 |   90
 21 | 14 |  120
 22 | 14 |   90
 23 | 14 |   90
 24 | 14 |  120
 25 | 14 |   90
 26 | 14 |   90
 27 | 14 |  120
 28 | 14 |   90
 29 | 14 |   90
  0 | 15 |  120
  1 | 15 |   90
  2 | 15 |   90
  3 | 15 |  120
  4 | 15 |   90
  5 | 15 |   90
  6 | 15 |  120
  7 | 15 |   90
  8 | 15 |   90
  9 | 15 |  120
 10 | 15 |   90
 11 | 15 |   90
 12 | 15 |  120
 13 | 15 |   90
 14 | 15 |   90
 15 | 15 |  240
 16 | 15 |  180
 17 | 15 |  180
 18 | 15 |  150
 19 | 15 |   90
 20 | 15 |   90
 21 | 15 |  120
 22 | 15 |   90
 23 | 15 |   90
 24 | 15 |  120
 25 | 15 |   90
 26 | 15 |   90
 27 | 15 |  120
 28 | 15 |   90
 29 | 15 |   90
  0 | 16 |  120
  1 | 16 |   90
  2 | 16 |   90
  3 | 16 |  120
  4 | 16 |   90
  5 | 16 |   90
  6 | 16 |  120
  7 | 16 |   90
  8 | 16 |   90
  9 | 16 |  120
 10 | 16 |   90
 11 | 16 |   90
 12 | 16 |  120
 13 | 16 |   90
 14 | 16 |   90
 15 | 16 |  120
 16 | 16 |   90
 17 | 16 |   90
 18 | 16 |  210
 19 | 16 |  180
 20 | 16 |  180
 21 | 16 |  120
 22 | 16 |   90
 23 | 16 |   90
 24 | 16 |  120
 25 | 16 |   90
 26 | 16 |   90
 27 | 16 |  120
 28 | 16 |   90
 29 | 16 |   90
  0 | 17 |  120
  1 | 17 |   90
  2 | 17 |   90
  3 | 17 |  120
  4 | 17 |   90
  5 | 17 |   90
  6 | 17 |  120
  7 | 17 |   90
  8 | 17 |   90
  9 | 17 |  120
 10 | 17 |   90
 11 | 17 |   90
 12 | 17 |  120
 13 | 17 |   90
 14 | 17 |   90
 15 | 17 |  120
 16 | 17 |   90
 17 | 17 |   90
 18 | 17 |  120
 19 | 17 |   90
 20 | 17 |   90
 21 | 17 |  240
 22 | 17 |  180
 23 | 17 |  180
 24 | 17 |  120
 25 | 17 |   90
 26 | 17 |   90
 27 | 17 |  120
 28 | 17 |   90
 29 | 17 |   90
  0 | 18 |  120
  1 | 18 |   90
  2 | 18 |   90
  3 | 18 |  120
  4 | 18 |   90
  5 | 18 |   90
  6 | 18 |  120
  7 | 18 |   90
  8 | 18 |   90
  9 | 18 |  120
 10 | 18 |   90
 11 | 18 |   90
 12 | 18 |  120
 13 | 18 |   90
 14 | 18 |   90
 15 | 18 |  120
 16 | 18 |   90
 17 | 18 |   90
 18 | 18 |  120
 19 | 18 |   90
 20 | 18 |   90
 21 | 18 |  120
 22 | 18 |   90
 23 | 18 |   90
 24 | 18 |  240
 25 | 18 |  180
 26 | 18 |  180
 27 | 18 |  150
 28 | 18 |   90
 29 | 18 |   90
  0 | 19 |  120
  1 | 19 |   90
  2 | 19 |   90
  3 | 19 |  120
  4 | 19 |   90
  5 | 19 |   90
  6 | 19 |  120
  7 | 19 |   90
  8 | 19 |   90
  9 | 19 |  120
 10 | 19 |   90
 11 | 19 |   90
 12 | 19 |  120
 13 | 19 |   90
 14 | 19 |   90
 15 | 19 |  120
 16 | 19 |   90
 17 | 19 |   90
 18 | 19 |  120
 19 | 19 |   90
 20 | 19 |   90
 21 | 19 |  120
 22 | 19 |   90
 23 | 19 |   90
 24 | 19 |  120
 25 | 19 |   90
 26 | 19 |   90
 27 | 19 |  210
 28 | 19 |  180
 29 | 19 |  180
  0 | 20 |  240
  1 | 20 |  180
  2 | 20 |  180
  3 | 20 |  120
  4 | 20 |   90
  5 | 20 |   90
  6 | 20 |  120
  7 | 20 |   90
  8 | 20 |   90
  9 | 20 |  120
 10 | 20 |   90
 11 | 20 |   90
 12 | 20 |  120
 13 | 20 |   90
 14 | 20 |   90
 15 | 20 |  120
 16 | 20 |   90
 17 | 20 |   90
 18 | 20 |  120
 19 | 20 |   90
 20 | 20 |   90
 21 | 20 |  120
 22 | 20 |   90
 23 | 20 |   90
 24 | 20 |  120
 25 | 20 |   90
 26 | 20 |   90
 27 | 20 |  120
 28 | 20 |   90
 29 | 20 |   90
  0 | 21 |  120
  1 | 21 |   90
  2 | 21 |   90
  3 | 21 |  240
  4 | 21 |  180
  5 | 21 |  180
  6 | 21 |  150
  7 | 21 |   90
  8 | 21 |   90
  9 | 21 |  120
 10 | 21 |   90
 11 | 21 |   90
 12 | 21 |  120
 13 | 21 |   90
 14 | 21 |   90
 15 | 21 |  120
 16 | 21 |   90
 17 | 21 |   90
 18 | 21 |  120
 19 | 21 |   90
 20 | 21 |   90
 21 | 21 |  120
 22 | 21 |   90
 23 | 21 |   90
 24 | 21 |  120
 25 | 21 |   90
 26 | 21 |   90
 27 | 21 |  120
 28 | 21 |   90
 29 | 21 |   90
  0 | 22 |  120
  1 | 22 |   90
  2 | 22 |   90
  3 | 22 |  120
  4 | 22 |   90
  5 | 22 |   90
  6 | 22 |  210
  7 | 22 |  180
  8 | 22 |  180
  9 | 22 |  120
 10 | 22 |   90
 11 | 22 |   90
 12 | 22 |  120
 13 | 22 |   90
 14 | 22 |   90
 15 | 22 |  120
 16 | 22 |   90
 17 | 22 |   90
 18 | 22 |  120
 19 | 22 |   90
 20 | 22 |   90
 21 | 22 |  120
 22 | 22 |   90
 23 | 22 |   90
 24 | 22 |  120
 25 | 22 |   90
 26 | 22 |   90
 27 | 22 |  120
 28 | 22 |   90
 29 | 22 |   90
  0 | 23 |  120
  1 | 23 |   90
  2 | 23 |   90
  3 | 23 |  120
  4 | 23 |   90
  5 | 23 |   90
  6 | 23 |  120
  7 | 23 |   90
  8 | 23 |   90
  9 | 23 |  240
 10 | 23 |  180
 11 | 23 |  180
 12 | 23 |  120
 13 | 23 |   90
 14 | 23 |   90
 15 | 23 |  120
 16 | 23 |   90
 17 | 23 |   90
 18 | 23 |  120
 19 | 23 |   90
 20 | 23 |   90
 21 | 23 |  120
 22 | 23 |   90
 23 | 23 |   90
 24 | 23 |  120
 25 | 23 |   90
 26 | 23 |   90
 27 | 23 |  120
 28 | 23 |   90
 29 | 23 |   90
  0 | 24 |  120
  1 | 24 |   90
  2 | 24 |   90
  3 | 24 |  120
  4 | 24 |   90
  5 | 24 |   90
  6 | 24 |  120
  7 | 24 |   90
  8 | 24 |   90
  9 | 24 |  120
 10 | 24 |   90
 11 | 24 |   90
 12 | 24 |  240
 13 | 24 |  180
 14 | 24 |  180
 15 | 24 |  150
 16 | 24 |   90
 17 | 24 |   90
 18 | 24 |  120
 19 | 24 |   90
 20 | 24 |   90
 21 | 24 |  120
 22 | 24 |   90
 23 | 24 |   90
 24 | 24 |  120
 25 | 24 |   90
 26 | 24 |   90
 27 | 24 |  120
 28 | 24 |   90
 29 | 24 |   90
  0 | 25 |  120
  1 | 25 |   90
  2 | 25 |   90
  3 | 25 |  120
  4 | 25 |   90
  5 | 25 |   90
  6 | 25 |  120
  7 | 25 |   90
  8 | 25 |   90
  9 | 25 |  120
 10 | 25 |   90
 11 | 25 |   90
 12 | 25 |  120
 13 | 25 |   90
 14 | 25 |   90
 15 | 25 |  210
 16 | 25 |  180
 17 | 25 |  180
 18 | 25 |  120
 19 | 25 |   90
 20 | 25 |   90
 21 | 25 |  120
 22 | 25 |   90
 23 | 25 |   90
 24 | 25 |  120
 25 | 25 |   90
 26 | 25 |   90
 27 | 25 |  120
 28 | 25 |   90
 29 | 25 |   90
  0 | 26 |  120
  1 | 26 |   90
  2 | 26 |   90
  3 | 26 |  120
  4 | 26 |   90
  5 | 26 |   90
  6 | 26 |  120
  7 | 26 |   90
  8 | 26 |   90
  9 | 26 |  120
 10 | 26 |   90
 11 | 26 |   90
 12 | 26 |  120
 13 | 26 |   90
 14 | 26 |   90
 15 | 26 |  120
 16 | 26 |   90
 17 | 26 |   90
 18 | 26 |  240
 19 | 26 |  180
 20 | 26 |  180
 21 | 26 |  120
 22 | 26 |   90
 23 | 26 |   90
 24 | 26 |  120
 25 | 26 |   90
 26 | 26 |   90
 27 | 26 |  120
 28 | 26 |   90
 29 | 26 |   90
  0 | 27 |  120
  1 | 27 |   90
  2 | 27 |   90
  3 | 27 |  120
  4 | 27 |   90
  5 | 27 |   90
  6 | 27 |  120
  7 | 27 |   90
  8 | 27 |   90
  9 | 27 |  120
 10 | 27 |   90
 11 | 27 |   90
 12 | 27 |  120
 13 | 27 |   90
 14 | 27 |   90
 15 | 27 |  120
 16 | 27 |   90
 17 | 27 |   90
 18 | 27 |  120
 19 | 27 |   90
 20 | 27 |   90
 21 | 27 |  240
 22 | 27 |  180
 23 | 27 |  180
 24 | 27 |  150
 25 | 27 |   90
 26 | 27 |   90
 27 | 27 |  120
 28 | 27 |   90
 29 | 27 |   90
  0 | 28 |  120
  1 | 28 |   90
  2 | 28 |   90
  3 | 28 |  120
  4 | 28 |   90
  5 | 28 |   90
  6 | 28 |  120
  7 | 28 |   90
  8 | 28 |   90
  9 | 28 |  120
 10 | 28 |   90
 11 | 28 |   90
 12 | 28 |  120
 13 | 28 |   90
 14 | 28 |   90
 15 | 28 |  120
 16 | 28 |   90
 17 | 28 |   90
 18 | 28 |  120
 19 | 28 |   90
 20 | 28 |   90
 21 | 28 |  120
 22 | 28 |   90
 23 | 28 |   90
 24 | 28 |  210
 25 | 28 |  180
 26 | 28 |  180
 27 | 28 |  120
 28 | 28 |   90
 29 | 28 |   90
  0 | 29 |  120
  1 | 29 |   90
  2 | 29 |   90
  3 | 29 |  120
  4 | 29 |   90
  5 | 29 |   90
  6 | 29 |  120
  7 | 29 |   90
  8 | 29 |   90
  9 | 29 |  120
 10 | 29 |   90
 11 | 29 |   90
 12 | 29 |  120
 13 | 29 |   90
 14 | 29 |   90
 15 | 29 |  120
 16 | 29 |   90
 17 | 29 |   90
 18 | 29 |  120
 19 | 29 |   90
 20 | 29 |   90
 21 | 29 |  120
 22 | 29 |   90
 23 | 29 |   90
 24 | 29 |  120
 25 | 29 |   90
 26 | 29 |   90
 27 | 29 |  240
 28 | 29 |  180
 29 | 29 |  180
(900 rows)

